tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -315,49 +315,45 @@ def extract_resource_id(policy, event):
     def run_for_event(policy, event=None):
         s = time.time()
 
-        with policy.ctx:
```

Comment:
The policy context is not designed to be entered more than once. Also here in event mode, we don't want to save output (which happens on context exit) if there are no resources that match the filter. It is not uncommon for the policy to be executed based on a trigger, but the filter expression excludes the resource. When that happens, we don't want to record the execution output. This change brings azure event mode behaviour into line with aws and gcp.